File: devops-scm-api/src/main/java/com/tencent/devops/scm/api/pojo/Tree.java
package com.tencent.devops.scm.api.pojo;

import com.tencent.devops.scm.api.enums.ContentKind;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Tree {
    private String path;
    private String sha;
    private String blobId;
    private ContentKind kind;
}

File: devops-scm-api/src/main/kotlin/com/tencent/devops/scm/api/pojo/Tree.kt
package com.tencent.devops.scm.api.pojo

import com.tencent.devops.scm.api.enums.ContentKind

data class Tree(
    // 文件路径
    val path: String,
    // SHA值
    val sha: String,
    // Blob ID
    val blobId: String,
    // 内容类型
    val kind: ContentKind
)
